WordPress/gutenberg
# ColorPalette: Improve color name readability

Fixes #48119


## What?
Improve readability of the color name and value by moving it off of the color swatch.

## Why?
See #48119 for more context and conversation, but in short having the text overlayed over the currently selected color can create contrast issues that are difficult to navigate due to the number of variables. One of these is that selecting a text color that will contrast well with the checkered pattern used to represent transparencies is likely going to be very difficult.

## How?

By moving the color name and value down below the swatch, contrast with the color itself and the transparency pattern are no longer a concern. The only contrast we'll need to worry about is the background color.

## Notes

- While the color name handles theming/dark mode well, the color value does not currently update based on theming changes. That said, the gray used currently is still visible when the current dark theme example is activated in Storybook.
- Similar to that, the border/box shadow does not update for darker backgrounds, and is pretty much lost on Storybook's dark theme example. I need to look more into both of these issues.
- I don't love how the popover changes size when a color is selected or cleared. In the current iteration, the name and value appear and disappear as needed. I'm wondering if it would make sense to leave that space visible at all times, and just leave it blank, or put some placeholder text like "Default color" in for the name until a value is selected. That way the sizing of the component could be consistent at all times.
- Potential followup: working on this has got me wondering about our goals with displaying the CSS variables as color values. Is that info helpful to the user? Would we be better off displaying the actual color instead?

## Testing Instructions
- View the component in Storybook, confirm that it shows the color data correctly and consistently. This includes the formatting and display of hex values vs. CSS variables.
- View the component in the editor (e.g. when changing text or background color of a paragraph block), confirm that it shows the color data correctly and consistently.

### Testing Instructions for Keyboard

- With `ColorPalette` open in Storybook or the editor, press `Tab` once to focus the picker button. Then press `Enter`/`Return` to select it.
- Use the arrow keys to move the slider and select a color.
- Press `Esc` to close the popover
- Press `Tab` once to confirm the focus moves to the first `CircleOptionPicker` option, confirming these changes don't alter the tab sequence.

### Screenshots


| Before  | After |
| ------------- | ------------- |
| ![image](https://github.com/WordPress/gutenberg/assets/13856531/3d353160-b1ac-458f-8f02-5017d4542a37) | ![Screenshot 2023-05-10 at 18 07 49](https://github.com/WordPress/gutenberg/assets/13856531/40d5eb0f-197d-4fba-ad16-9b3b73116afa) |
